### PR TITLE
Remove slf4j-log4j12 from Transport dependency graph

### PR DESCRIPTION
**Pull request by shardulm94**
Transport as a library should only expose `slf4j-api` to its consumers and should not expose any SLF4J bindings.

http://www.slf4j.org/faq.html#maven2
> Thus, as far as your users are concerned you are exporting slf4j-api as a transitive dependency of your library, but not any SLF4J-binding or any underlying logging system.

Unfortunately `hadoop-common` does expose `sl4fj-log4j12` as a transitive. This can cause `StackOverflowError` if another binding is also present on the classpath. This issue seems to popup frequently in Presto tests and on the Presto cluster. We as a library, should remove the bindings from our transitives and ensure we don't introduce any ourselves.

Note that we still keep the `slf4j-simple` binding for tests and explictly for spark test framework as Spark Core and Spark SQL do not include a binding.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -48,11 +48,20 @@ subprojects {
     strictCheck true
   }
 
+  configurations {
+    all {
+      exclude group: 'org.slf4j', module: 'slf4j-log4j12'
```
Comment (wmoustafa) [inline build.gradle:53]: Could you please add a comment with the first part of the PR description here? Would help with future maintenance.

File: build.gradle [hunk 2]
```diff
@@ -48,11 +48,20 @@ subprojects {
...
+  }
+
   plugins.withType(JavaPlugin) {
     project.apply(plugin: 'checkstyle')
 
     dependencies {
+      compile 'org.slf4j:slf4j-api:1.7.25'
+
       testCompile 'org.testng:testng:6.11'
+      testCompile 'org.slf4j:slf4j-simple:1.7.25'
```
Comment (wmoustafa) [inline build.gradle:64]: Is this required by all modules? I think it is required for the plugin. Depending on the answer of this, should we relocate the above section as well?

Comment (shardulm94) [reply build.gradle:64]: I think we should include the `slf4j-simple` binding by default in our test compile path for all modules. This allows any dependencies which use `slf4j-api` to correctly print out logging statements during test cases even though Transport explicitly maybe not be using it.
Yes `slf4j-api` is probably not needed by all modules, but I thought I would just keep it here. ideally we would have logging statements in all our modules and most libraries have `slf4j-api` exported in their dependency graph anyway so it being there is not a surprise. Let me know if you would like to change that.

File: build.gradle (hunk 1 above)
Comment (shardulm94) [reply build.gradle:59]: Done

File: build.gradle (hunk 2 above)
Comment (shardulm94) [reply build.gradle]: Removed `slf4j-api` turns out we don't need it. We have no log lines. The only one in TransportPlugin uses Gradle's logging libraries.

**Pull request merged**